packages/lit-html/src/lit-html.ts

```diff
@@ -945,6 +958,8 @@ class ChildPart {
   private _textSanitizer: ValueSanitizer | undefined;
   /** @internal */
   _$parent: Disconnectable | undefined;
+  /** @internal */
+  __isConnected = true;
```

Comment:
Should this be initialized via options?

Reply:
Why is this distinct from this.options' `isConnected`?

Reply:
The connection state of the top-level part (the part created/returned by `render` that the user can call `part.setConnection(true/false)` on) is the source-of-truth for everything below it. The `options._$isConnected` flag is not the source of truth, but rather a communication channel between the top-level part and everything below it since they all get references to this one options bag passed down.

> Should this be initialized via options?

In theory, the `__isConnected` flag on any parts below the top-level part returned is never used (and the top-level part should never get an `options` bag with `_$isConnected` in it), so it doesn't really matter, but we could keep it in sync just for consistency.

Reply:
I think there's a flaw though... It's possible to get a reference to a child part that shares `options` with the top-level part, and then call `part.setConnected(false)` and change the connection state. From there, if you did `part.setValue(...)`, directives would still see the top-level connection state, not the local one.

Reply:
Per design conversation, re-worked to use `get _$isConnected() { return this._$parent._$isConnected }` on all Disconnectables up to the `RootChildPart`, which holds the state for the entire tree.

Reply:
And we're TODO'ing setting the initial connected state in render because in LitElement it's almost always going to be first rendered when connected, yeah?

Reply:
Yeah, added TODO's, a skipped failing test for this (https://github.com/lit/lit/pull/2034/files#diff-fc790d086c06e1adeb4c976a60945f0b9b50aa86350a9c9c6b61618c48bac95dR522), and issue https://github.com/lit/lit/issues/2051